unittests/backends/oqc/OQCTester.cpp

```diff
@@ -0,0 +1,181 @@
+  auto backendString = fmt::format(fmt::runtime(backendStringTemplate),
+                                   mockPort, email, password);
+
+  auto &platform = cudaq::get_platform();
+  platform.setTargetBackend(backendString);
+
+  auto [kernel, theta] = cudaq::make_kernel<double>();
+  auto qubit = kernel.qalloc(2);
+  kernel.x(qubit[0]);
+  kernel.ry(theta, qubit[1]);
+  kernel.x<cudaq::ctrl>(qubit[1], qubit[0]);
+
+  using namespace cudaq::spin;
+  cudaq::spin_op h = 5.907 - 2.1433 * x(0) * x(1) - 2.1433 * y(0) * y(1) +
+                     .21829 * z(0) - 6.125 * z(1);
+  auto result = cudaq::observe(kernel, h, .59);
+  result.dump();
+
+  printf("ENERGY: %lf\n", result.exp_val_z());
+  EXPECT_NEAR(result.exp_val_z(), -1.7, 1e-1);
+}
+
+CUDAQ_TEST(OQCTester, checkObserveAsync) {
+  auto backendString = fmt::format(fmt::runtime(backendStringTemplate),
+                                   mockPort, email, password);
+
+  auto &platform = cudaq::get_platform();
+  platform.setTargetBackend(backendString);
+
+  auto [kernel, theta] = cudaq::make_kernel<double>();
+  auto qubit = kernel.qalloc(2);
+  kernel.x(qubit[0]);
+  kernel.ry(theta, qubit[1]);
+  kernel.x<cudaq::ctrl>(qubit[1], qubit[0]);
+
+  using namespace cudaq::spin;
+  cudaq::spin_op h = 5.907 - 2.1433 * x(0) * x(1) - 2.1433 * y(0) * y(1) +
+                     .21829 * z(0) - 6.125 * z(1);
+  auto future = cudaq::observe_async(kernel, h, .59);
+
+  auto result = future.get();
+  result.dump();
+
+  printf("ENERGY: %lf\n", result.exp_val_z());
+  EXPECT_NEAR(result.exp_val_z(), -1.7, 1e-1);
+}
+
+CUDAQ_TEST(OQCTester, checkObserveAsyncLoadFromFile) {
+  auto backendString = fmt::format(fmt::runtime(backendStringTemplate),
+                                   mockPort, email, password);
+
+  auto &platform = cudaq::get_platform();
+  platform.setTargetBackend(backendString);
+
+  auto [kernel, theta] = cudaq::make_kernel<double>();
+  auto qubit = kernel.qalloc(2);
+  kernel.x(qubit[0]);
+  kernel.ry(theta, qubit[1]);
+  kernel.x<cudaq::ctrl>(qubit[1], qubit[0]);
+
+  using namespace cudaq::spin;
+  cudaq::spin_op h = 5.907 - 2.1433 * x(0) * x(1) - 2.1433 * y(0) * y(1) +
+                     .21829 * z(0) - 6.125 * z(1);
+  auto future = cudaq::observe_async(kernel, h, .59);
+
+  {
+    std::ofstream out("saveMeObserve.json");
+    out << future;
+  }
+
+  // Later you can come back and read it in
+  cudaq::async_result<cudaq::observe_result> readIn(&h);
+  std::ifstream in("saveMeObserve.json");
+  in >> readIn;
+
+  // Get the results of the read in future.
+  auto result = readIn.get();
+
+  std::remove("saveMeObserve.json");
+  result.dump();
+
+  printf("ENERGY: %lf\n", result.exp_val_z());
+  EXPECT_NEAR(result.exp_val_z(), -1.7, 1e-1);
+}
+
+int main(int argc, char **argv) {
+  out << "key: key\nrefresh: refresh\ntime: 0";
+  out.close();
+  ::testing::InitGoogleTest(&argc, argv);
+  auto ret = RUN_ALL_TESTS();
+  std::remove(fileName.c_str());
```

Comment:
Lines 175, 176, and 179 cause this to not compile for me. Are those intentional?

Reply:
Those look stray to me - i'll remove them.

Reply:
It looks like the `std::remove(filename` has snuck back in with this latest update.